hooks/command

```diff
@@ -84,24 +115,46 @@ download_binary_and_run() {
   get_architecture || return 1
   local _arch="$RETVAL"
   local _executable="monorepo-diff-buildkite-plugin"
+  local _executable_version_file="monorepo-diff-buildkite-plugin.version"
   local _repo="https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin"
 
   get_version || return 1
-  local _version="$RETVAL"
+  local _specified_version="$RETVAL"
+  local _latest_version=""
+  local _binary_version=""
+  local test_mode="${BUILDKITE_PLUGIN_MONOREPO_DIFF_BUILDKITE_PLUGIN_TEST_MODE:-false}"
+  local _url=""
 
-  if [ -z "${_version}" ]; then
-    _url=${_repo}/releases/latest/download/${_executable}_${_arch}
-  else
-    _url=${_repo}/releases/download/${_version}/${_executable}_${_arch}
+  if check_cmd "./${_executable}"; then
+    if check_binary_version; then
+      _binary_version="$RETVAL"
+    else
+      say "Warning: Could not determine binary version, will download fresh copy"
+    fi
```

Comment:
with my other suggestion this would need to change to:

```suggestion
     _binary_version=$(get_binary_version)
     if [[ -z "$_binary_version" ]]; then
       say "Warning: Could not determine binary version, will download fresh copy"
     fi
```